packages/server/graphql/resolvers/resolveThread.ts

```diff
@@ -0,0 +1,55 @@
+import {Threadable} from '../../database/types/Threadable'
+import TaskDB from '../../database/types/Task'
+
+const resovleThread = async ({id: threadSourceId}, _args, {dataLoader}) => {
+  const [comments, tasks] = await Promise.all([
+    dataLoader.get('commentsByThreadId').load(threadSourceId),
+    dataLoader.get('tasksByThreadId').load(threadSourceId)
+  ])
```

Comment:
we don't pass `threadSource` here to further qualify the query. might be possible for an `agendaItemId` and a `reflectionGroupId` to collide?

Reply:
great observation!
today, all database IDs are created via `shortid`, which is guaranteed to be unique as long as they are all generated by the same server. as we scale horizontally, we'll need to invest some time to keep this guarantee true, but for now it's safe to say that across all rethinkdb tables there are no shared IDs.